lib/experimental/fonts-library/class-wp-rest-fonts-library-controller.php

```diff
@@ -0,0 +1,215 @@
+ *
+ * This file contains the class for the REST API Fonts Library Controller.
+ *
+ * @package    Gutenberg
+ * @subpackage Fonts Library
+ * @since      X.X.X
+ */
+
+if ( class_exists( 'WP_REST_Fonts_Library_Controller' ) ) {
+	return;
+}
+
+/**
+ * Fonts Library Controller class.
+ */
+class WP_REST_Fonts_Library_Controller extends WP_REST_Controller {
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->rest_base = 'fonts_library';
+		$this->namespace = 'wp/v2';
+	}
+
+	/**
+	 * Registers the routes for the objects of the controller.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => WP_REST_Server::EDITABLE,
+					'callback'            => array( $this, 'install_fonts' ),
+					'permission_callback' => array( $this, 'update_fonts_library_permissions_check' ),
+				),
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => WP_REST_Server::DELETABLE,
+					'callback'            => array( $this, 'uninstall_font_family' ),
+					'permission_callback' => array( $this, 'update_fonts_library_permissions_check' ),
+				),
+			)
+		);
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/google_fonts',
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_google_fonts' ),
+					'permission_callback' => array( $this, 'read_fonts_library_permissions_check' ),
+				),
+			)
+		);
+
+	}
+
+	/**
+	 * Removes a font family from the fonts library and all their assets.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function uninstall_font_family( $request ) {
+		$data = array(
+			'slug' => $request['slug'],
+		);
+		$font = new WP_Font_Family( $data );
+		$result = $font->uninstall();
+
+		if ( is_wp_error( $result ) ) {
+			return $result;
+		}
+
+		return new WP_REST_Response( __( 'Font family uninstalled successfully.', 'gutenberg' ), 200 );
+	}
+
+	/**
+	 * Check if user has permissions to update the fonts library.
+	 *
+	 * @return true|WP_Error True if the request has write access for the item, WP_Error object otherwise.
+	 */
+	public function update_fonts_library_permissions_check() {
+		if ( ! current_user_can( 'edit_theme_options' ) ) {
+			return new WP_Error(
+				'rest_cannot_update_fonts_library',
+				__( 'Sorry, you are not allowed to update the fonts library on this site.', 'gutenberg' ),
+				array(
+					'status' => rest_authorization_required_code(),
+				)
+			);
+		}
+
+		// The update endpoints requires write access to the temp and the fonts directories.
+		$temp_dir = get_temp_dir();
+		if ( ! is_writable( $temp_dir ) || ! wp_is_writable( WP_FONTS_DIR ) ) {
```

Comment:
`get_temp_dir()` is documented to return a writable directory, so the first condition shouldn't be necessary here.

> **get_temp_dir()**
> Determines a writable directory for temporary files.
> [Ref](https://developer.wordpress.org/reference/functions/get_temp_dir/)

Reply:
With the current syntax, we are checking if `/tmp/` is writable and we use just `get_temp_dir()` we would not check that. cases where `/tmp/` is not writable should be rare but I think the check could remain.

Reply:
Yep, no harm in being extra safe :slightly_smiling_face: